controller/controller.js

```diff
@@ -1,16 +1,98 @@
-// c = controller, m = model, v = view, db = database
 const model = require("../model/model");
-
-// TODO: find out how error catching is meant to be, repeating .catch here seems not very DRY
+const errors = require("../controller/errors.js");
 
 // /api/topics Endpoints
-exports.getApiTopics = (request, response) => {
-    model.selectTopics().then((selectTopicsResponseObject) => {
+const getApiTopics = (request, response, next) => {
+    model.topics.selectTopics().then((selectTopicsResponseObject) => {
         response.status(200).send(selectTopicsResponseObject);
     }).catch((err) => {
-        console.log(err);
+        next(err);
+    });
+};
+
+// /api/articles Endpoints
+const getApiArticles = (request, response, next) => {
+    model.articles.selectArticles().then((responseObject) => {
+
+        response.status(200).send(responseObject);
+
+    }).catch((err) => {
+        next(err);
+    });
+};
+
+// /api/articles/:article_id Endpoints
+const getArticleById = (request, response, next) => {
+    const articleId = request.params["article_id"];
+
+    model.articles.selectArticleByArticleId(articleId).then((responseObject) => {
+          response.status(200).send(responseObject);
+    }).catch((err) => {
+        next(err);
+    });
+
+};
+
+// 6. GET /api/articles/:article_id/comments
+const getCommentsByArticleId = (request, response, next) => {
+    const articleId = request.params["article_id"];
+
+    model.comments.selectCommentsByArticleId(articleId).then((responseObject) => {
+
+   
+        response.status(200).send(responseObject);
+    }).catch((err) => {
+        next(err);
+      });
+
+};
+
+// 7. POST /api/articles/:article_id/comments
+const postCommentToArticle = (request, response, next) => {
+    const articleId = request.params["article_id"];
```

Comment:
While you are being consistent with accessing the article_id using square bracket notation its not necessary you can still access with dot notation :)